File: go/credit_note.go
// this file is @generated
package meteroid

import "time"

type CreditNote struct {
	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	CreditNoteNumber string `json:"credit_note_number"`

	CreditType CreditType `json:"credit_type"`

	CreditedAmountCents int64 `json:"credited_amount_cents"`

	Currency Currency `json:"currency"`

	// User-defined custom property values, keyed by definition `key`.
	CustomProperties RequiredMap[any] `json:"custom_properties"`

	CustomerId CustomerId `json:"customer_id"`

	// RFC 3339 timestamp.
	FinalizedAt *time.Time `json:"finalized_at,omitempty"`

	Id CreditNoteId `json:"id"`

	InvoiceId InvoiceId `json:"invoice_id"`

	InvoiceNumber string `json:"invoice_number"`

	LineItems RequiredSlice[InvoiceLineItem] `json:"line_items"`

	Memo *string `json:"memo,omitempty"`

	PlanVersionId *PlanVersionId `json:"plan_version_id,omitempty"`

	Reason *string `json:"reason,omitempty"`

	RefundedAmountCents int64 `json:"refunded_amount_cents"`

	Status CreditNoteStatus `json:"status"`

	SubscriptionId *SubscriptionId `json:"subscription_id,omitempty"`

	Subtotal int64 `json:"subtotal"`

	TaxAmount int64 `json:"tax_amount"`

	TaxBreakdown RequiredSlice[TaxBreakdownItem] `json:"tax_breakdown"`

	Total int64 `json:"total"`

	// RFC 3339 timestamp.
	UpdatedAt *time.Time `json:"updated_at,omitempty"`

	// RFC 3339 timestamp.
	VoidedAt *time.Time `json:"voided_at,omitempty"`
}

File: python/meteroid/models/credit_note.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .credit_note_id import CreditNoteId
from .credit_note_status import CreditNoteStatus
from .credit_type import CreditType
from .currency import Currency
from .customer_id import CustomerId
from .invoice_id import InvoiceId
from .invoice_line_item import InvoiceLineItem
from .plan_version_id import PlanVersionId
from .subscription_id import SubscriptionId
from .tax_breakdown_item import TaxBreakdownItem


@dataclasses.dataclass
class CreditNote(BaseModel):
    created_at: datetime

    credit_note_number: str

    credit_type: CreditType

    credited_amount_cents: int

    currency: Currency

    custom_properties: t.Any
    """User-defined custom property values, keyed by definition `key`."""

    customer_id: CustomerId

    id: CreditNoteId

    invoice_id: InvoiceId

    invoice_number: str

    line_items: t.List[InvoiceLineItem]

    refunded_amount_cents: int

    status: CreditNoteStatus

    subtotal: int

    tax_amount: int

    tax_breakdown: t.List[TaxBreakdownItem]

    total: int

    finalized_at: t.Optional[datetime] = None

    memo: t.Optional[str] = None

    plan_version_id: t.Optional[PlanVersionId] = None

    reason: t.Optional[str] = None

    subscription_id: t.Optional[SubscriptionId] = None

    updated_at: t.Optional[datetime] = None

    voided_at: t.Optional[datetime] = None
